src/layouts/NavLayout/index.tsx

```diff
@@ -1,17 +1,148 @@
+  const [isSidebarOpen, setisSidebarOpen] = useState(false);
+  const router = useRouter();
+  const cardsData = useQuery({
+    queryKey: [KEY.CARDS],
+    queryFn: () => getCards(),
+  });
+
+  const collectionData = useQuery({
+    queryKey: [KEY.COLLECTION],
+    queryFn: () => getCollection(),
+  });
+
+  const { pathname } = useRouter();
+
+  useEffect(() => {
+    setisSidebarOpen(false);
+  }, [router.pathname]);
 
+  if (pathname === '/cards') {
+    return (
+      <>
+        <NavBar
+          leadingButton="menu"
+          trailingButton="add"
+          onClickLeft={() => setisSidebarOpen(!isSidebarOpen)}
+          onClickRight={() => router.push('/generation')}
+        />
+        {isSidebarOpen &&
+          (userState.name ? (
+            <SideBar
+              isLogined={true}
+              name={userState.name}
+              loginIcon={userState.oauthServerType}
+              myCardCnt={cardsData.data?.length}
+              collectCardCnt={collectionData.data?.length}
+              onClose={() => setisSidebarOpen(!isSidebarOpen)}
+              onClickLogout={() => {}}
+            />
+          ) : (
+            <SideBar
+              isLogined={false}
+              onClickLogin={() => router.push('/login')}
+              onClose={() => setisSidebarOpen(!isSidebarOpen)}
+            />
+          ))}
+        <ContentLayout>{children}</ContentLayout>
+      </>
+    );
+  } else if (pathname === '/generation') {
+    return (
+      <>
+        <NavBar
+          leadingButton="back"
+          onClickLeft={() => {
+            router.back();
+          }}
+        />
+        <ContentLayout>{children}</ContentLayout>
+      </>
+    );
+  } else if (pathname === '/cards/[id]') {
+    return (
+      <>
+        <NavBar
+          leadingButton="back"
+          trailingButton="flip"
+          onClickLeft={() => {
+            router.back();
+          }}
+          onClickRight={() => {}}
+        />
+        <ContentLayout>{children}</ContentLayout>
+      </>
+    );
+  } else if (pathname === '/cards/[id]/edit') {
+    return (
+      <>
+        <NavBar
+          title="명함 수정하기"
+          leadingButton="back"
+          onClickLeft={() => {
+            router.back();
+          }}
+        />
+        <ContentLayout>{children}</ContentLayout>
+      </>
+    );
+  } else if (pathname === '/settings') {
+    return (
+      <>
+        <NavBar
+          title="환경설정"
+          leadingButton="back"
+          onClickLeft={() => {
+            router.back();
+          }}
+        />
+        <ContentLayout>{children}</ContentLayout>
+      </>
+    );
+  }
   return (
-    <div>
-      {userState.name ? <NavBar /> : <div>로그인안됨</div>}
-      {children}
-    </div>
+    <>
+      <NavBar
+        leadingButton="menu"
+        showSearchBar={true}
+        onClickLeft={() => setisSidebarOpen(!isSidebarOpen)}
+      />
+      {isSidebarOpen &&
+        (userState.name ? (
+          <SideBar
```

Comment:
```suggestion
    <>
      <NavBar
        leadingButton="menu"
        trailingButton={pathname === '/cards' ? 'add' : null}
        showSearchBar={true}
        onClickLeft={() => setisSidebarOpen(!isSidebarOpen)}
      />
      {isSidebarOpen &&
        (userState.name ? (
          <SideBar
```
이 하단 부분이 거의 pathname === '/cards` 와 같아 보이는데요. 위처럼 내부에서 분기처리를 하면 코드 중복을 최소화할 수 있지 않을까요?